doc/build.md

```diff
@@ -291,3 +291,15 @@ Unit tests can be disable by adding -DDISABLE_TESTS=1 to CMake.
 ```shell
 -DDISABLE_TESTS=1
 ```
+
+### Embedded builds for RISC-V

```

Comment:
Thanks to put this section. 

1) We have a section 7 - `Supported Architecture and Cross-Compiler based on X64 platform` in https://github.com/DMTF/libspdm/blob/main/README.md#features.

You may find: [RISCV_GNU](https://github.com/riscv/riscv-gnu-toolchain), [RISCV64_GCC](https://packages.ubuntu.com/bionic/gcc-riscv64-linux-gnu),
[RISCV_XPACK](https://github.com/xpack-dev-tools/riscv-none-elf-gcc-xpack)

Could you please add a similar item there? RISCV_NONE.

2) In https://github.com/DMTF/libspdm/blob/main/doc/build.md#compiler-for-riscv32riscv64-choose-one, we describe how to get the compiler.

Could you please add a section on how to get this `riscv64-elf-gcc` ?

3) For the name, why you call `riscv64-elf-gcc` as `RISCV-none` ? But not `RISCV64-ELF-GCC` ?

Reply:
I added 1 and 2.

For the name, it matches the other names. You have `RISCV_GNU` for using the GNU tools (like glibc), `RISCV64_GCC` for using GCC and `RISCV_XPACK` for using xpack. I thought `RISCV_NONE` describes that we aren't using a libc to create the binaries.

I'm happy to change it. I could go with `RISCV_NOSTDLIB` if that's better?

Reply:
I checked riscv64-elf-gcc in ArchLinux - "The GNU Compiler Collection - cross compiler for RISCV64 (bare-metal) target" at  https://archlinux.org/packages/extra/x86_64/riscv64-elf-gcc/

I suggest `RISCV64_NONE`.

Reply:
I have changed the URL to `https://gcc.gnu.org/` to indicate that it's just upstream GCC. This isn't Arch specific so I don't want to point to the Arch package.

I think `RISCV64_NONE` is confusing as it supports both 32 and 64 bit RISC-V. In fact it probably will be used exclusively for 32-bit as that's what embedded systems commonly are

Reply:
> I have changed the URL to `https://gcc.gnu.org/` to indicate that it's just upstream GCC. This isn't Arch specific so I don't want to point to the Arch package.

That is confusing. In build document, you have below:
```
d) [RISCV NONE](https://gcc.gnu.org/)
  - `sudo pacman -Syu riscv32-elf-binutils riscv32-elf-newlib riscv64-elf-binutils riscv64-elf-gcc riscv64-elf-newlib`
```

NOTE: `pacman` is only for Arch linux.
If it is generic, then we need a better way to describe it.
For example, I am still using ubuntu, does it support?

Reply:
Ok, I have updated the docs to describe what is required

It should work on Ubuntu, you will just need to install the required toolchain

Reply:
I got failure on ubuntu. Please help me understand how to.

```
jyao1@jyao1-desk:~$ sudo apt-get install riscv32-elf-binutils riscv32-elf-newlib riscv64-elf-binutils riscv64-elf-gcc riscv64-elf-newlib
[sudo] password for jyao1:
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package riscv32-elf-binutils
E: Unable to locate package riscv32-elf-newlib
E: Unable to locate package riscv64-elf-binutils
E: Unable to locate package riscv64-elf-gcc
E: Unable to locate package riscv64-elf-newlib
```

Reply:
> I think RISCV64_NONE is confusing as it supports both 32 and 64 bit RISC-V. 

My original goal is to let the compiler name match the compiler binary. Since the compiler name has `riscv64-`, I dont think anything wrong to use `RISCV64_` as prefix.

> In fact it probably will be used exclusively for 32-bit as that's what embedded systems commonly are

I am not sure what you mean - `exclusively for 32-bit`. Would you please elaborate a little bit?

Reply:
> My original goal is to let the compiler name match the compiler binary. Since the compiler name has `riscv64-`, I dont think anything wrong to use `RISCV64_` as prefix.

Hmmm... I see what you mean. I just find it really confusing to call it `RISCV64_NONE` when it supports 32/64-bit RISC-V

> I am not sure what you mean - `exclusively for 32-bit`. Would you please elaborate a little bit?

This is targeting an embedded system not running Linux. That is pretty much only going to be 32-bit systems